include/mxnet/base.h

```diff
@@ -316,6 +321,19 @@ inline Context Context::GPU(int32_t dev_id) {
   return Create(kGPU, dev_id);
 }
 
+inline int32_t Context::GetGPUCount() {
+#if MXNET_USE_CUDA
+  int32_t count;
+  cudaError_t e = cudaGetDeviceCount(&count);
+  if (e != cudaSuccess) {
+    return 0;
```

Comment:
This should result in an error in front end. Use use cuda_call checks

Reply:
The reason I did it like this is that on a CPU host, where I would expect to get a device count of 0, I got error 30 when calling cudaGetDeviceCount. This is probably why we have the same logic in [storage.cc](https://github.com/apache/incubator-mxnet/blob/121adb6660897d7cef223caba80a43412ab5c9e0/src/storage/storage.cc#L104).

Reply:
that said, it is probably indeed cleaner to raise an exception in case cuda reports one.

Reply:
Any idea why cuda reports error? Maybe we can special case error 30 and raise the other errors?

Reply:
no, I'm not sure why this happens. The error message and error code are unfortunately not very informative. 

With respect to CUDA_CALL, unfortunately it is defined in `src/common/cuda_utils.h` so that I can't import it from `base.h`.

It is probably cleanest to bubble up any CUDA errors, so I changed the code to do so accordingly to your suggestion.